### PR TITLE
Introduce ArchiveOperations.gzip(path) and bzip2(path) 

### PR DESCRIPTION
**Pull request by eskatos**
This is a follow up to in-flight work at https://github.com/gradle/gradle/pull/15433
Actually reverting that PR and preferring the introduction of new functions on the existing `ArchiveOperations` service.

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java [hunk 1]
```diff
@@ -28,6 +29,16 @@ public DefaultArchiveOperations(FileOperations fileOperations) {
         this.fileOperations = fileOperations;
     }
 
+    @Override
+    public ReadableResource gzip(Object path) {
+        return fileOperations.getResources().gzip(path);
```
Comment (bamboo) [inline subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java:34]: 💄 my Demeter OCD would have me extract `fileOperations.getResources()` to a private `resources()` method :)

**Review by bamboo — APPROVED** [review]
LGTM!

File: subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java (hunk 1 above)
Comment (eskatos) [reply subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java]: OCD satisfied :)

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java (hunk 1 above)
Comment (bamboo) [reply subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultArchiveOperations.java]: Thank you! :D

**Pull request merged**